File: src/main/scala/org/openapitools/server/model/CategoryDTO.scala
package org.openapitools.server.model


/**
 * Информация о категории.  Категория считается листовой, если у нее нет дочерних категорий. 
 *
 * @param id Идентификатор категории. for example: ''null''
 * @param name Название категории. for example: ''null''
 * @param children Дочерние категории. for example: ''null''
*/
final case class CategoryDTO (
  id: Long,
  name: String,
  children: Option[Seq[CategoryDTO]] = None
)

File: ympa_fsharp_functions_server/src/model/CategoryDTO.fs
namespace ympa_fsharp_functions_server.Model

open System
open System.Collections.Generic
open Newtonsoft.Json

module CategoryDTO =

  //#region CategoryDTO

  [<CLIMutable>]
  type CategoryDTO = {
    [<JsonProperty(PropertyName = "id")>]
    Id : int64;
    [<JsonProperty(PropertyName = "name")>]
    Name : string;
    [<JsonProperty(PropertyName = "children")>]
    Children : CategoryDTO[];
  }

  //#endregion
